homeassistant/components/meater/__init__.py

```diff
@@ -0,0 +1,92 @@
+"""The Meater Temperature Probe integration."""
+from datetime import timedelta
+import logging
+
+import async_timeout
+from meater import (
+    AuthenticationError,
+    MeaterApi,
+    ServiceUnavailableError,
+    TooManyRequestsError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Meater Temperature Probe from a config entry."""
+    # Store an API object to access
+    session = async_get_clientsession(hass)
+    meater_api = MeaterApi(session)
+
+    # Add the credentials
+    try:
+        _LOGGER.debug("Authenticating with the Meater API")
+        await meater_api.authenticate(
+            entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD]
+        )
+    except (ServiceUnavailableError, TooManyRequestsError) as err:
+        raise ConfigEntryNotReady from err
+    except AuthenticationError as err:
+        _LOGGER.error("Unable to authenticate with the Meater API: %s", err)
+        return False
```

Comment:
Should we raise a `ConfigEntryAuthFailed` here instead?
(Not 100% if we can, as this PR doesn't implement a reauth flow).

It would, however, display the error message on the integrations dashboard

Reply:
Let's add a reauth flow in a follow-up 👍